.github/workflows/e2e-main-tf.yaml

```diff
@@ -0,0 +1,145 @@
+# Copyright (C) 2025 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: PD E2E Testing Farm
+
+on:
+  push:
+    branches: [main]
+    
+  workflow_dispatch:
+    inputs:
+      fork:
+        default: 'podman-desktop'
+        description: 'Podman Desktop repo fork'
+        type: string
+        required: true
+      branch:
+        default: 'main'
+        description: 'Podman Desktop repo branch'
+        type: string
+        required: true
+      podman_version:
```

Comment:
Not blocking request: Open an issue to provide an option to use a podman version which is derived from Podman Desktop codebase (in podman extension we are hardcoding a podman version to be provided with podman-desktop) which should be default podman version used in testing.